src/cwe_checker_lib/src/pcode/term.rs

```diff
@@ -543,11 +636,12 @@ impl Project {
     /// according to the program headers of the binary.
     pub fn into_ir_project(self, binary_base_address: u64) -> IrProject {
         let mut program: Term<IrProgram> = Term {
-            tid: self.program.tid,
+            tid: self.program.tid.clone(),
             term: self
                 .program
                 .term
-                .into_ir_program(binary_base_address, self.stack_pointer_register.size),
+                .clone()
+                .into_ir_program(binary_base_address, &self),
```

Comment:
This cloning is quite expensive (since the IR program should be a very large data structure for large binaries). I think we should try to get rid of the `clone()`.